contribs/gnodev/pkg/events/middleware.go

```diff
@@ -0,0 +1,105 @@
+package events
+
+import (
+	"bytes"
+	_ "embed"
+	"net/http"
+	"strings"
+	"sync"
+	"text/template"
+)
+
+//go:embed static/hotreload.js
+var reloadscript string
+
+type Middleware struct {
+	remote   string
+	muRemote sync.RWMutex
+
+	next     http.Handler
+	tmpl     *template.Template
+	onceExec *sync.Once
+	script   []byte
+}
+
+func NewMiddleware(remote string, next http.Handler) *Middleware {
+	tmpl := template.Must(template.New("reloadscript").
+		Funcs(tmplFuncs).
+		Parse(reloadscript))
+
+	return &Middleware{
+		tmpl:     tmpl,
+		remote:   remote,
+		next:     next,
+		onceExec: &sync.Once{},
+	}
+}
+
+type middlewareResponseWriter struct {
+	http.ResponseWriter
+	buffer *bytes.Buffer
+}
+
+func (m *middlewareResponseWriter) Write(b []byte) (int, error) {
+	return m.buffer.Write(b)
+}
+
+func (m *Middleware) UpdateRemote(remote string) {
+	m.muRemote.Lock()
+	m.remote = remote
+	m.onceExec = &sync.Once{}
+	m.muRemote.Unlock()
+}
```

Comment:
Unused?

Reply:
i've forgot about this one, I originally made this to handle the possibility of having `:0` as listening address for the node, so i can dynamically update the remote address if needed. Since im not using it, I will stash this somewhere, and maybe implement it later.